src/helm/proxy/clients/palmyra_client.py

```diff
@@ -0,0 +1,133 @@
+            "topP": request.top_p,
+            "bestOf": request.top_k_per_token,
+            "stop": request.stop_sequences,
+            # random_seed have been disabled for now.
+            # It is here to ensure that Writer does not cache the request when we
+            # want several completions with the same prompt. Right now it seems
+            # to have no effect so we are disabling it.
+            # TODO(#1515): re-enable it when it works.
+            # "random_seed": request.random,
+        }
+
+        completions: List[Sequence] = []
+        model_name: str = request.model_engine
+
+        # `num_completions` is not supported, so instead make `num_completions` separate requests.
+        for completion_index in range(request.num_completions):
+            try:
+                # This is disabled for now. See above TODO(#1515).
+                # HACKY: Use the random seed to get different results for each completion.
+                # raw_request["random_seed"] = (
+                #     f"completion_index={completion_index}"
+                #     if request.random is None
+                #     else request.random + f":completion_index={completion_index}"
+                # )
+
+                def do_it():
+                    result = self._send_request(model_name, raw_request)
+                    if "choices" not in result:
+                        raise ValueError(f"Invalid response: {result}")
+                    return result
+
+                # We need to include the engine's name to differentiate among requests made for different model
+                # engines since the engine name is not included in the request itself.
+                # In addition, we want to make `request.num_completions` fresh
+                # requests, cache key should contain the completion_index.
+                # Echoing the original prompt is not officially supported by Writer. We instead prepend the
+                # completion with the prompt when `echo_prompt` is true, so keep track of it in the cache key.
+                cache_key = Client.make_cache_key(
+                    {
+                        "engine": request.model_engine,
+                        "completion_index": completion_index,
+                        **raw_request,
+                    },
+                    request,
+                )
+
+                response, cached = self.cache.get(cache_key, wrap_request_time(do_it))
+            except (requests.exceptions.RequestException, AssertionError) as e:
+                error: str = f"PalmyraClient error: {e}"
+                return RequestResult(success=False, cached=False, error=error, completions=[], embedding=[])
+
+            # Post process the completion.
+            response_text: str = response["choices"][0]["text"]
+            # NOTE(josselin): The Writer API sometimes returns a space at the beginning of the completion.
+            # We remove it here, but this is a hacky solution.
```

Comment:
We shouldn't need this right? The leading space gets removed downstream in HELM e.g. [here](https://github.com/stanford-crfm/helm/blob/c36c5f900de99e2e26418af3923ce7dffecdd74f/src/helm/benchmark/metrics/basic_metrics.py#L116)